t/ui/12-needle-edit.t

```diff
@@ -518,10 +518,22 @@ subtest 'Showing new needles limited to the 5 most recent ones' => sub {
       or diag explain \@needle_names;
 };
 
+subtest 'Broken needle is handled gracefully' => sub {
+    # Change the JSON to something invalid
+    $needle_json_file->spurt("{{,};");
+    $driver->refresh();
+    $driver->title_is('openQA: Needle Editor', 'needle editor still shows up');
+    is(
+        $driver->find_element('#editor_warnings span')->get_text(),
+        join("\n", 'Could not parse needle: inst-timezone-text for opensuse 13.1', @expected_needle_warnings),
+        'warning about deleted needle displayed (beside new needle warnings)'
+    );
+};
+
 subtest 'Deletion of needle is handled gracefully' => sub {
     # delete needle on disk and reload the needle editor
     $needle_json_file->remove;
-    $driver->get($driver->get_current_url);
+    $driver->refresh();
```

Comment:
Changes like that should be at least in a separate commit.

Reply:
#2884